tuna/miopen/driver/base.py

```diff
@@ -103,6 +108,20 @@ def get_common_cols() -> List[str]:
     """Returns common MIOpenDriver command line args"""
     return ['wall', 'time', 'iter', 'verify']
 
+  def parse_fdb_key(self, line: str) -> None:
+    """import config attributes from fdb key line"""
+    fds: dict
+    direction: str
+    fds, _, direction = get_fds_from_cmd(line)
+    setattr(self, 'direction', DIR_MAP[direction])
+    for key in self.to_dict():
+      if key in fds:
+        setattr(self, key, fds[key])
+
+    pattern_3d = '[0-9]x[0-9]x[0-9]'
+    if search(pattern_3d, line):
+      setattr(self, 'spatial_dim', 3)
```

Comment:
What if the line has patterns like `3x256x256`? Would this be more general:
```python
pattern_3d = "[0-9]+x[0-9]+x[0-9]+"
```

Reply:
Also, maybe @cderb  can answer this better: what if the line has all patterns like `1x256x256`. Then the spatial dim should be 2, right?

Reply:
> What if the line has patterns like `3x256x256`? Would this be more general:
> 
> ```python
> pattern_3d = "[0-9]+x[0-9]+x[0-9]+"
> ```

This is correct, this regex needs '+' to accommodate for larger numbers.
If the spatial dim is 2 the fdb key should not include the depth field.

Reply:
Also why did this function move to base.py from convolution.py?